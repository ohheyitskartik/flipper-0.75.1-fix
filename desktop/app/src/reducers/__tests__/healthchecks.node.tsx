/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 */

import {
  default as reducer,
  startHealthchecks,
  finishHealthchecks,
  updateHealthcheckResult,
  acknowledgeProblems,
} from '../healthchecks';
import {Healthchecks, EnvironmentInfo} from 'flipper-doctor';

const HEALTHCHECKS: Healthchecks = {
  ios: {
    label: 'iOS',
    isSkipped: false,
    isRequired: true,
    healthchecks: [
      {
        key: 'ios.sdk',
        label: 'SDK Installed',
        run: async (_env: EnvironmentInfo) => {
          return {hasProblem: false, message: ''};
        },
      },
    ],
  },
  android: {
    label: 'Android',
    isSkipped: false,
    isRequired: true,
    healthchecks: [
      {
        key: 'android.sdk',
        label: 'SDK Installed',
        run: async (_env: EnvironmentInfo) => {
          return {hasProblem: true, message: 'Error'};
        },
      },
    ],
  },
  common: {
    label: 'Common',
    isSkipped: false,
    isRequired: false,
    healthchecks: [
      {
        key: 'common.openssl',
        label: 'OpenSSL Istalled',
        run: async (_env: EnvironmentInfo) => {
          return {hasProblem: false, message: ''};
        },
      },
    ],
  },
};

test('startHealthCheck', () => {
  const res = reducer(undefined, startHealthchecks(HEALTHCHECKS));
  expect(res).toMatchSnapshot();
});

test('updateHealthcheckResult', () => {
  let res = reducer(undefined, startHealthchecks(HEALTHCHECKS));
  res = reducer(
    res,
    updateHealthcheckResult('android', 'android.sdk', {
      message: 'Updated Test Message',
      isAcknowledged: false,
      status: 'SUCCESS',
    }),
  );
  expect(res).toMatchSnapshot();
});

test('finish', () => {
  let res = reducer(undefined, startHealthchecks(HEALTHCHECKS));
  res = reducer(
    res,
    updateHealthcheckResult('ios', 'ios.sdk', {
      message: 'Updated Test Message',
      isAcknowledged: false,
      status: 'SUCCESS',
    }),
  );
  res = reducer(
    res,
    updateHealthcheckResult('android', 'android.sdk', {
      message: 'Updated Test Message',
      isAcknowledged: false,
      status: 'SUCCESS',
    }),
  );
  res = reducer(
    res,
    updateHealthcheckResult('common', 'common.openssl', {
      message: 'Updated Test Message',
      isAcknowledged: false,
      status: 'SUCCESS',
    }),
  );
  res = reducer(res, finishHealthchecks());
  expect(res).toMatchSnapshot();
});

test('statuses updated after healthchecks finished', () => {
  let res = reducer(undefined, startHealthchecks(HEALTHCHECKS));
  res = reducer(
    res,
    updateHealthcheckResult('android', 'android.sdk', {
      message: 'Updated Test Message',
      isAcknowledged: false,
      status: 'FAILED',
    }),
  );
  res = reducer(
    res,
    updateHealthcheckResult('ios', 'ios.sdk', {
      message: 'Updated Test Message',
      isAcknowledged: false,
      status: 'SUCCESS',
    }),
  );
  res = reducer(
    res,
    updateHealthcheckResult('common', 'common.openssl', {
      message: 'Updated Test Message',
      isAcknowledged: false,
      status: 'SUCCESS',
    }),
  );
  res = reducer(res, finishHealthchecks());
  expect(res).toMatchSnapshot();
});

test('acknowledgeProblems', () => {
  let res = reducer(undefined, startHealthchecks(HEALTHCHECKS));
  res = reducer(
    res,
    updateHealthcheckResult('ios', 'ios.sdk', {
      isAcknowledged: false,
      status: 'FAILED',
    }),
  );
  res = reducer(
    res,
    updateHealthcheckResult('android', 'android.sdk', {
      isAcknowledged: false,
      status: 'SUCCESS',
    }),
  );
  res = reducer(
    res,
    updateHealthcheckResult('common', 'common.openssl', {
      isAcknowledged: false,
      status: 'FAILED',
    }),
  );
  res = reducer(res, finishHealthchecks());
  res = reducer(res, acknowledgeProblems());
  expect(res).toMatchSnapshot();
});
