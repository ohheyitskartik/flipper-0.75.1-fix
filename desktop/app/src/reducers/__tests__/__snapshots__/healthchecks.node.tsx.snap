// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`acknowledgeProblems 1`] = `
Object {
  "acknowledgedProblems": Array [
    "ios.sdk",
    "common.openssl",
  ],
  "healthcheckReport": Object {
    "categories": Object {
      "android": Object {
        "checks": Object {
          "android.sdk": Object {
            "key": "android.sdk",
            "label": "SDK Installed",
            "result": Object {
              "isAcknowledged": true,
              "status": "SUCCESS",
            },
          },
        },
        "key": "android",
        "label": "Android",
        "result": Object {
          "isAcknowledged": true,
          "status": "SUCCESS",
        },
      },
      "common": Object {
        "checks": Object {
          "common.openssl": Object {
            "key": "common.openssl",
            "label": "OpenSSL Istalled",
            "result": Object {
              "isAcknowledged": true,
              "status": "FAILED",
            },
          },
        },
        "key": "common",
        "label": "Common",
        "result": Object {
          "isAcknowledged": true,
          "status": "FAILED",
        },
      },
      "ios": Object {
        "checks": Object {
          "ios.sdk": Object {
            "key": "ios.sdk",
            "label": "SDK Installed",
            "result": Object {
              "isAcknowledged": true,
              "status": "FAILED",
            },
          },
        },
        "key": "ios",
        "label": "iOS",
        "result": Object {
          "isAcknowledged": true,
          "status": "FAILED",
        },
      },
    },
    "result": Object {
      "isAcknowledged": true,
      "status": "FAILED",
    },
  },
}
`;

exports[`finish 1`] = `
Object {
  "acknowledgedProblems": Array [],
  "healthcheckReport": Object {
    "categories": Object {
      "android": Object {
        "checks": Object {
          "android.sdk": Object {
            "key": "android.sdk",
            "label": "SDK Installed",
            "result": Object {
              "isAcknowledged": false,
              "message": "Updated Test Message",
              "status": "SUCCESS",
            },
          },
        },
        "key": "android",
        "label": "Android",
        "result": Object {
          "isAcknowledged": false,
          "status": "SUCCESS",
        },
      },
      "common": Object {
        "checks": Object {
          "common.openssl": Object {
            "key": "common.openssl",
            "label": "OpenSSL Istalled",
            "result": Object {
              "isAcknowledged": false,
              "message": "Updated Test Message",
              "status": "SUCCESS",
            },
          },
        },
        "key": "common",
        "label": "Common",
        "result": Object {
          "isAcknowledged": false,
          "status": "SUCCESS",
        },
      },
      "ios": Object {
        "checks": Object {
          "ios.sdk": Object {
            "key": "ios.sdk",
            "label": "SDK Installed",
            "result": Object {
              "isAcknowledged": false,
              "message": "Updated Test Message",
              "status": "SUCCESS",
            },
          },
        },
        "key": "ios",
        "label": "iOS",
        "result": Object {
          "isAcknowledged": false,
          "status": "SUCCESS",
        },
      },
    },
    "result": Object {
      "status": "SUCCESS",
    },
  },
}
`;

exports[`startHealthCheck 1`] = `
Object {
  "acknowledgedProblems": Array [],
  "healthcheckReport": Object {
    "categories": Object {
      "android": Object {
        "checks": Object {
          "android.sdk": Object {
            "key": "android.sdk",
            "label": "SDK Installed",
            "result": Object {
              "status": "IN_PROGRESS",
            },
          },
        },
        "key": "android",
        "label": "Android",
        "result": Object {
          "status": "IN_PROGRESS",
        },
      },
      "common": Object {
        "checks": Object {
          "common.openssl": Object {
            "key": "common.openssl",
            "label": "OpenSSL Istalled",
            "result": Object {
              "status": "IN_PROGRESS",
            },
          },
        },
        "key": "common",
        "label": "Common",
        "result": Object {
          "status": "IN_PROGRESS",
        },
      },
      "ios": Object {
        "checks": Object {
          "ios.sdk": Object {
            "key": "ios.sdk",
            "label": "SDK Installed",
            "result": Object {
              "status": "IN_PROGRESS",
            },
          },
        },
        "key": "ios",
        "label": "iOS",
        "result": Object {
          "status": "IN_PROGRESS",
        },
      },
    },
    "result": Object {
      "status": "IN_PROGRESS",
    },
  },
}
`;

exports[`statuses updated after healthchecks finished 1`] = `
Object {
  "acknowledgedProblems": Array [],
  "healthcheckReport": Object {
    "categories": Object {
      "android": Object {
        "checks": Object {
          "android.sdk": Object {
            "key": "android.sdk",
            "label": "SDK Installed",
            "result": Object {
              "isAcknowledged": false,
              "message": "Updated Test Message",
              "status": "FAILED",
            },
          },
        },
        "key": "android",
        "label": "Android",
        "result": Object {
          "isAcknowledged": false,
          "status": "FAILED",
        },
      },
      "common": Object {
        "checks": Object {
          "common.openssl": Object {
            "key": "common.openssl",
            "label": "OpenSSL Istalled",
            "result": Object {
              "isAcknowledged": false,
              "message": "Updated Test Message",
              "status": "SUCCESS",
            },
          },
        },
        "key": "common",
        "label": "Common",
        "result": Object {
          "status": "SUCCESS",
        },
      },
      "ios": Object {
        "checks": Object {
          "ios.sdk": Object {
            "key": "ios.sdk",
            "label": "SDK Installed",
            "result": Object {
              "isAcknowledged": false,
              "message": "Updated Test Message",
              "status": "SUCCESS",
            },
          },
        },
        "key": "ios",
        "label": "iOS",
        "result": Object {
          "status": "SUCCESS",
        },
      },
    },
    "result": Object {
      "isAcknowledged": false,
      "status": "FAILED",
    },
  },
}
`;

exports[`updateHealthcheckResult 1`] = `
Object {
  "acknowledgedProblems": Array [],
  "healthcheckReport": Object {
    "categories": Object {
      "android": Object {
        "checks": Object {
          "android.sdk": Object {
            "key": "android.sdk",
            "label": "SDK Installed",
            "result": Object {
              "isAcknowledged": false,
              "message": "Updated Test Message",
              "status": "SUCCESS",
            },
          },
        },
        "key": "android",
        "label": "Android",
        "result": Object {
          "status": "IN_PROGRESS",
        },
      },
      "common": Object {
        "checks": Object {
          "common.openssl": Object {
            "key": "common.openssl",
            "label": "OpenSSL Istalled",
            "result": Object {
              "status": "IN_PROGRESS",
            },
          },
        },
        "key": "common",
        "label": "Common",
        "result": Object {
          "status": "IN_PROGRESS",
        },
      },
      "ios": Object {
        "checks": Object {
          "ios.sdk": Object {
            "key": "ios.sdk",
            "label": "SDK Installed",
            "result": Object {
              "status": "IN_PROGRESS",
            },
          },
        },
        "key": "ios",
        "label": "iOS",
        "result": Object {
          "status": "IN_PROGRESS",
        },
      },
    },
    "result": Object {
      "status": "IN_PROGRESS",
    },
  },
}
`;
